server/controllers/UserController.js

```diff
@@ -37,41 +38,56 @@ exports.sendMessage = async (parent, args, context) => {
     let room = await Room.findOne({
       _id: args.roomId,
       members: { $in: [context.currentUser.id] },
-    });
+    }).populate("members");
 
     if (!room) {
       throw new ApolloError(
         "Room not found or you are not a member of this room"
       );
     }
 
+    // parse mentions
+    const mentions = parseMentions(args.content);
+
+    // check if mentioned users are member of the room
+    const mentioned_users = mentions
+      .map(m => {
+        let found = room.members.find(i => i.username === m);
+        if (found) {
+          return found._id;
+        }
+        return null;
+      })
+      // remove null values & current user if mentioned
+      .filter(userId => {
+        if (!userId) return false;
+        return `${userId}` !== `${context.currentUser.id}`;
+      });
+
```

Comment:
parsing the mentions and extracting the _ids, i know i can use usernames directly but i wanted to be able to populate the "mentions" if needed (not currently needed though)